components/eam/src/physics/cam/gw_common.F90

```diff
@@ -742,4 +746,1472 @@ subroutine gw_drag_prof(ncol, ngwv, src_level, tend_level, do_taper, dt, &
+  IF (gsd_gwd_bl.and.(ls_taper .GT. 1.E-02)) THEN
+
+  do i = its,ite
+    if(.not.ldrag(i)) then
+  !
+  !------- determine the height of flow-blocking layer
+  !
+        kblk = 0
+        pe = 0.0_r8
+
+        do k = kte, kpblmin, -1
+          if(kblk.eq.0 .and. k.le.komax(i)) then
+            !flow block appears within the reference level
+            !compare potential energy and kinetic energy
+            !divided by g*ro is to turn del(pa) into height
+            pe = pe + bnv2(i,k)*(zl(i,komax(i))-zl(i,k))*del(i,k)/g/ro(i,k)
+            ke = 0.5_r8*((rcs*u1(i,k))**2._r8+(rcs*v1(i,k))**2._r8)
+  !
+  !---------- apply flow-blocking drag when pe >= ke 
+  !
+            if(pe.ge.ke) then
+              kblk = k
+              kblk = min(kblk,kbl(i))
+              zblk = zl(i,kblk)-zl(i,kts)
+            endif
+          endif
+        enddo
+
+        if(kblk.ne.0) then
+  !
+  !--------- compute flow-blocking stress
+  !
+
+          !dxmax_ls is different than the usual one
+          !because the taper is very different
+          !dxy is a length scale mostly in the direction of the flow to the ridge
+          !so it is good and not needed for an uneven grid area
+          !ref Lott and Miller (1997) original scheme
+          cd = max(2.0_r8-1.0_r8/od(i),0.0_r8)
+          !
+          !tuning of the drag magnitude
+          cd=ncd*cd
+          !
+          taufb(i,kts) = 0.5_r8 * roll(i) * coefm(i) / max(dxmax_ls,dxy(i))**2 * cd * dxyp(i)   &
+                         * olp(i) * zblk * ulow(i)**2
+          !changed grid box area into dy*dy
+          tautem = taufb(i,kts)/float(kblk-kts)
+          do k = kts+1, kblk
+            taufb(i,k) = taufb(i,k-1) - tautem
+          enddo
+
+          !
+          !----------sum orographic GW stress and flow-blocking stress
+          !
+          !taup(i,:) = taup(i,:) + taufb(i,:)   ! Keep taup and taufb separate for now
+        endif
+    endif
+  enddo
+
+  ENDIF   ! end blocking drag
+!===========================================================
+  IF (gsd_gwd_ls.OR.gsd_gwd_bl.and.(ls_taper .GT. 1.E-02)) THEN
+  !                                                                       
+  !  calculate - (g)*d(tau)/d(pressure) and deceleration terms dtaux, dtauy
+  !
+
+   do k = kts,kte
+     do i = its,ite
+       taud_ls(i,k) = 1._r8 * (taup(i,k+1) - taup(i,k)) * csg / del(i,k)
+       taud_bl(i,k) = 1._r8 * (taufb(i,k+1) - taufb(i,k)) * csg / del(i,k)
+     enddo
+   enddo
+  !                                                                       
+  !  limit de-acceleration (momentum deposition ) at top to 1/2 value 
+  !  the idea is some stuff must go out the 'top'                     
+  !                                                                       
+
+   do klcap = lcap,kte
+     do i = its,ite
+       taud_ls(i,klcap) = taud_ls(i,klcap) * factop
+       taud_bl(i,klcap) = taud_bl(i,klcap) * factop
+     enddo
+   enddo
```

Comment:
It seems lcap is hard coded to be equal to kte, is that intentional?

Reply:
It is hardcoded, intentional.